packages/react-components/react-components/src/AccessibilityExamples/FormErrorsMessages.stories.tsx

```diff
@@ -0,0 +1,305 @@
+                always: () => {
+                  if (!formState.isSubmitting) {
+                    formValidation.onFieldValidated(fullNameId);
+                  }
+                  return true;
+                },
+              },
+            }}
+          />
+          {errors.fullName?.types && (
+            <ValidationMessage id={fullNameId} formValidation={formValidation}>
+              {'required' in errors.fullName.types ? (
+                <p>Full name is required.</p>
+              ) : (
+                <>
+                  <p>Full name is invalid. It must:</p>
+                  <ul>
+                    {('minLength' in errors.fullName.types || 'maxLength' in errors.fullName.types) && (
+                      <li>Have between 2 and 50 characters.</li>
+                    )}
+                    {'onlyNameChars' in errors.fullName.types && (
+                      <li>Contain only lowercase or uppercase letters, spaces or hyphens.</li>
+                    )}
+                    {'startsAndEndsWithLetter' in errors.fullName.types && <li>Start and end wit letter.</li>}
+                  </ul>
+                </>
+              )}
+            </ValidationMessage>
+          )}
+
+          <Label htmlFor={emailId}>E-mail</Label>
+          <Controller
+            name="email"
+            control={control}
+            as={
+              <Input
+                type="text"
+                id={emailId}
+                aria-required="true"
+                aria-invalid={!!errors.email}
+                aria-describedby={variant == 'bad' ? undefined : `${emailId}Errors`}
+              />
+            }
+            rules={{
+              required: true,
+              validate: {
+                validEmail: value => regexes.validEmail.test(value),
+                always: () => {
+                  if (!formState.isSubmitting) {
+                    formValidation.onFieldValidated(emailId);
+                  }
+                  return true;
+                },
+              },
+            }}
+          />
+          {errors.email?.types && (
+            <ValidationMessage id={emailId} formValidation={formValidation}>
+              {'required' in errors.email.types ? (
+                <p>E-mail is required.</p>
+              ) : (
+                <>
+                  <p>E-mail is invalid. It must:</p>
+                  <ul>
+                    {'validEmail' in errors.email.types && <li>Be a valid e-mail address, like name@example.com.</li>}
+                  </ul>
+                </>
+              )}
+            </ValidationMessage>
+          )}
+
+          <Button type="submit">Subscribe</Button>
+        </form>
+      ) : (
+        <p id={`${variant}-validMessage`} role="alert" tabIndex={-1}>
+          The form is valid and would have been submitted.
+        </p>
+      )}
+    </>
+  );
+};
+
+export const FormErrorMessagesExample = () => {
```

Comment:
Is this different from the Form validation Accessibility scenario? I am wondering if we should just add information from here to the existing scenario

Reply:
The accessibility scenario form behaves in the same manner. The difference is that here for simplicity we use only two edit fields (full name and email) whereas in the scenario we use a more complex form. We cannot simply reuse or link to the scenario here, because here we need to demonstrate that omitting the "aria-describedby" attribute is an error.